File: 1926.cpp
#include <iostream>
#define X first
#define Y second
#include <queue>
using namespace std;

int board[502][502];
bool vis[502][502];
int n, m; // n은 행의 수, m은 열의 수
int dx[4] = {1, 0, -1, 0};
int dy[4] = {0, 1, 0, -1};
int main(){
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cin >> n >> m;
    for (int i = 0; i < n; i++){
        for (int j = 0; j < m; j++) cin >> board[i][j];
    }
    int mx = 0; // 그림의 최댓값
    int num = 0; // 그림의 수
    for (int i = 0; i < n; i++){
        for (int j = 0; j < m; j++){
            if(board[i][j] == 0 || vis[i][j]) continue;
            num++;
            queue<pair<int, int> > Q;
            vis[i][j] = 1; // 시작점을 방문했다고 명시
            Q.push({i, j}); // 큐에 시작점을 삽입
            int area = 0;
            while(!Q.empty()){
                area++;
                pair<int, int> cur = Q.front(); Q.pop();
                for (int dir = 0; dir < 4; dir++){
                    int nx = cur.X + dx[dir];
                    int ny = cur.Y + dy[dir];
                    if(nx < 0 || nx >= n || ny < 0 || ny >= m) continue;
                    if(vis[nx][ny] || board[nx][ny] != 1) continue;
                    vis[nx][ny] = 1;
                    Q.push({nx, ny});
                }
            }
            mx = max(mx, area);
        }
    }
    cout << num << '\n' << mx;
    return 0;
}

File: PY/1926.py
import sys
from collections import deque

input = sys.stdin.readline
n, m = map(int, input().split()) # 행, 열

board = []
for _ in range(n):
    board.append(list(map(int, input().split())))

vis = [[False for _ in range(m)] for _ in range(n)]

dx = [1, 0, -1, 0]
dy = [0, 1, 0, -1]

mx = 0  # 그림의 최댓값
num = 0 # 그림의 수

for i in range(n):
    for j in range(m):
        if board[i][j] == 0 or vis[i][j]:
            continue
        num += 1
        Q = deque()
        vis[i][j] = True
        Q.append((i, j))
        area = 0
        while Q:
            area += 1
            cur = Q.popleft()
            for dir in range(4):
                nx = cur[0] + dx[dir]
                ny = cur[1] + dy[dir]

                if nx < 0 or nx >= n or ny < 0 or ny >= m:
                    continue

                if vis[nx][ny] or board[nx][ny] != 1:
                    continue

                vis[nx][ny] = True   
                Q.append((nx, ny))
            
            mx = max(mx, area)

print(num)
print(mx)